test/LanguageServer.Engine.Tests/TestProject.cs

```diff
@@ -0,0 +1,386 @@
+        /// </summary>
+        /// <param name="testClassType">
+        ///     The type of test-suite class used to determine the assembly containing the currently-running test (which can be used to find the test's deployment directory).
+        /// </param>
+        /// <param name="relativePathSegments">
+        ///     The project file's relative (to the test deployment directory) path segments.
+        /// </param>
+        /// <returns>
+        ///     The loaded project, as a <see cref="TestProject"/>.
+        /// </returns>
+        public static TestProject LoadTestProject(this ProjectCollection projectCollection, Type testClassType, params string[] relativePathSegments)
+        {
+            if (projectCollection == null)
+                throw new ArgumentNullException(nameof(projectCollection));
+
+            if (testClassType == null)
+                throw new ArgumentNullException(nameof(testClassType));
+
+            if (relativePathSegments == null)
+                throw new ArgumentNullException(nameof(relativePathSegments));
+
+            string projectFileName = GetProjectFile(testClassType, relativePathSegments);
+
+            string projectFileContent = File.ReadAllText(projectFileName);
+            XmlDocumentSyntax projectXml = Parser.ParseText(projectFileContent);
+
+            var xmlPositions = new TextPositions(projectFileContent);
+            var xmlLocator = new XmlLocator(projectXml, xmlPositions);
+
+            Project msbuildProject = projectCollection.GetLoadedProjects(projectFileName).FirstOrDefault();
+            if (msbuildProject == null)
+                msbuildProject = projectCollection.LoadProject(projectFileName);
+
+            var msbuildObjectLocator = new MSBuildObjectLocator(msbuildProject, xmlLocator, xmlPositions);
+
+            return new TestProject(msbuildProject, msbuildObjectLocator, projectXml, xmlLocator, xmlPositions);
+        }
+
+        /// <summary>
+        ///     Load a test project's XML.
+        /// </summary>
+        /// <typeparam name="TTestClass">
+        ///     The test-suite class used to determine the assembly containing the currently-running test (which can be used to find the test's deployment directory).
+        /// </typeparam>
+        /// <param name="relativePathSegments">
+        ///     The project file's relative (to the test deployment directory) path segments.
+        /// </param>
+        /// <returns>
+        ///     The loaded project, as <see cref="TestProjectXml"/>.
+        /// </returns>
+        public static TestProjectXml LoadXml<TTestClass>(params string[] relativePathSegments)
+            where TTestClass : class
+        {
+            if (relativePathSegments == null)
+                throw new ArgumentNullException(nameof(relativePathSegments));
+
+            Type testClassType = typeof(TTestClass);
+
+            return LoadXml(testClassType, relativePathSegments);
```

Comment:
```suggestion
            return LoadXml(typeof(TTestClass), relativePathSegments);
```
Same as one of reviews above